### PR TITLE
Added Null check for Git hub issue#417

### PR DESCRIPTION
**Pull request by kiran-vadla**
Hi Pablo, 

Please review this pull request and merge . This was created for https://github.com/oasp/oasp4j/issues/417.

Thanks,
Kiran.

### CODE REVIEW

**Comment by maybeec** [comment]
Unfortunately, I am not convinced that we should merge this as we would null-check our internal program variants here. This is simply overhead. Any further opinions?

**Comment by kiran-vadla** [comment]
Hi All, 

Any further opinions on this ? 

Thanks,
Kiran.

**Comment by hohwille** [comment]
I agree with @maybeec.
OrderPosition can not live by itself and hence always needs a parent Order.
Hence, we need to change DDL and add a non-null constraint on the orderId column.
This is still not a fix for #417 but that will let us find the root cause of the problem why an OrderPosition gets created with orderId being null.

**Comment by hohwille** [comment]
BTW: convention is to ommit "is null" in ```Objects.requireNonNull``` check messages. Just specify the name of the field being null.

**Comment by kiran-vadla** [comment]
@hohwille , We already have a Not Null check on RESTAURANTORDER table in the DDL. 

Thanks,
Kiran.

**Comment by kiran-vadla** [comment]
@hohwille , I mean on the Id column of RESTAURANTORDER  table

**Comment by kiran-vadla** [comment]
@hohwille , Please ignore my above comments . I overlooked your statement . I am going to add NOT NULL check for the column order_id in ORDERPOSITION table and run 'mvn clean install' to see if there are any issues .Thanks , Kiran.

**Comment by kiran-vadla** [comment]
@hohwille , I have added NOT NULL constraint on ORDER_ID column in ORDERPOSITION table in V0001__R001_Create_schema.sql and executed 'mvn clean install' and see no issues or errors. So, submitted these changes to the branch . I can see the changes reflecting in the PR. Please review the PR once again.

File: samples/core/src/main/java/io/oasp/gastronomy/restaurant/salesmanagement/logic/impl/usecase/UcManageOrderPositionImpl.java [hunk 1]
```diff
@@ -96,6 +96,7 @@ public OrderPositionEto saveOrderPosition(OrderPositionEto orderPosition) {
     if (orderPositionId == null) {
       action = "saved";
       Long offerId = orderPosition.getOfferId();
+      Objects.requireNonNull(offerId, "offerId cannot be null");
```
Comment (hohwille) [inline samples/core/src/main/java/io/oasp/gastronomy/restaurant/salesmanagement/logic/impl/usecase/UcManageOrderPositionImpl.java]: can you please remove the " connot be null" here.

**Comment by hohwille** [comment]
Good improvement. Still does not fix the bug #417 as we need to find the root of the problem but this constraint is fine to accept as improvement. Could you also improve the null check so we can merge this PR?

**Comment by hohwille** [comment]
@maybeec 
> Unfortunately, I am not convinced that we should merge this as we would null-check our internal program variants here. This is simply overhead.

I disagree. This is a REST service and we can not know the callee for sure and we can never trust him. So we should properly check input. Maybe BeanValidation would be a better choice than explicit code with null checks but checking is fine.

**Comment by kiran-vadla** [comment]
@hohwille , I have made the changes (Remove the string "cannot be null" from Null check) as you have mentioned and pushed the changes to the branch . Please review.

**Comment by hohwille** [comment]
@kiran-vadla :+1: thanks for the quick improvments. awesome.

**Pull request merged**